frame/system/src/lib.rs

```diff
@@ -923,6 +928,44 @@ impl<T: Trait> Module<T> {
 	}
 }
 
+impl<T: Trait> Module<T> {
+	pub fn spent_weight() -> Option<Weight> {
+		CurrentDispatchableWeight::get()
+	}
+
+	/// Decrease the amount of weight consumed by the current dispatchable.
+	///
+	/// This function can only meaningfully called within `Self::dispatch`. Otherwise,
+	/// it does nothing and returns `Err`.
+	///
+	/// The total weight returned from the dispatchable cannot exceed its total requested weight.
+	pub fn return_unspent_weight(unspent: Weight) -> Result<(), ()> {
+		let weight = match CurrentDispatchableWeight::take() {
+			None => return Err(()),
+			Some(weight) => weight,
+		};
+		let next_weight = weight.checked_sub(unspent).ok_or(())?;
+		CurrentDispatchableWeight::put(next_weight);
+		Ok(())
+	}
+
+	/// Dispatch a given `T::Call` with a given origin.
+	///
+	/// Returns how much weight was unspent.
+	pub fn dispatch_call(
```

Comment:
```suggestion
	pub fn note_dispatch_call(
```
There should really be an emphasis on the fact that this some bookkeeping for you. Chose some other name if you want, but  `dispatch_call ` is too simple.